### ISSUE TITLE
TypeScript version – Branch available :white_check_mark:

### ISSUE THREAD

**Issue by eritislami** [eritislami/evobot#1167]
Hello All,

I have published a branch for the brand new TypeScript version of the bot, this should be the last "big" change at least for the near future, I closed some issues that have been fixed in this branch like the stop command/reaction breaking the music queue-ing.

I would appreciate it if everyone that reads this tried to use the new version as I won't be fixing bugs or adding new things to the previous js-only code, I will try to keep this branch up for some time before I merge it to master branch.

https://github.com/eritislami/evobot/tree/refactor/typescript

**Comment by nikosszzz** [comment]
I've merged it into my bot. Seems like it is working and easy to be modified to each person's use.

**Comment by nikosszzz** [comment]
Also: it seems like it doesnt deploy to heroku correctly. 

![image](https://user-images.githubusercontent.com/53112175/170980844-22301d96-166f-4dcc-92c9-cfcba440ebe3.png)
This error pops up only on heroku, meanwhile locally running the bot is fine.

**Comment by eritislami** [comment]
@nikosszzz you can run `npm run build` to have it compiled and ready to go, after you can run the index in /dist folder

**Comment by nikosszzz** [comment]
Yeah, I got it sorted. Is soundcloud support coming in the next few days? I saw a commented out scdl line in one of the structs.

**Comment by nikosszzz** [comment]
![image](https://user-images.githubusercontent.com/53112175/173060888-0d5334b0-4e78-43b4-9a48-d607df32d757.png)
The search command throws this error when the time to choose a video rans out. Code is the same as on the repo for the catch error part.

```
} catch (error: any) {
            console.error(error);
            (message.channel as CustomTextChannel).activeCollector = false;
            message.channel.send(`Time to choose a video ran out.`).catch(console.error);
        };
    },
```

**Comment by eritislami** [comment]
> I only changed the message to the channel that is sent to that instead of `error.message`, as to silence that error temporarily.

I put the common error as the reply for now, thanks

**Comment by SmegDan** [comment]
I get this error when trying to run it.

TSError: ⨯ Unable to compile TypeScript:
utils/i18n.ts:1:18 - error TS7016: Could not find a declaration file for module 'i18n'. '/app/node_modules/i18n/index.js' implicitly has an 'any' type.
